x/stakeibc/keeper/msg_server_clear_balance_test.go

```diff
@@ -0,0 +1,116 @@
+
+	"github.com/Stride-Labs/stride/x/stakeibc/types"
+	stakeibctypes "github.com/Stride-Labs/stride/x/stakeibc/types"
+)
+
+type ClearBalanceState struct {
+	feeChannel Channel
+}
+
+type ClearBalanceTestCase struct {
+	initialState ClearBalanceState
+	validMsg     stakeibctypes.MsgClearBalance
+}
+
+func (s *KeeperTestSuite) SetupClearBalance() ClearBalanceTestCase {
+	// fee account
+	feeAccountOwner := fmt.Sprintf("%s.%s", HostChainId, "FEE")
+	feeChannelID := s.CreateICAChannel(feeAccountOwner)
+	feeAddress := s.IcaAddresses[feeAccountOwner]
+	// hz
+	zoneAddress := types.NewZoneAddress(HostChainId)
+	hostZone := stakeibctypes.HostZone{
+		ChainId:        HostChainId,
+		ConnectionId:   ibctesting.FirstConnectionID,
+		HostDenom:      Atom,
+		IBCDenom:       IbcAtom,
+		RedemptionRate: sdk.NewDec(1.0),
+		Address:        zoneAddress.String(),
+		FeeAccount: &stakeibctypes.ICAAccount{
+			Address: feeAddress,
+			Target:  stakeibctypes.ICAAccountType_FEE,
+		},
+	}
+
+	amount := uint64(1_000_000)
+
+	user := Account{
+		acc: s.TestAccs[0],
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	return ClearBalanceTestCase{
+		initialState: ClearBalanceState{
+			feeChannel: Channel{
+				PortID:    icatypes.PortPrefix + feeAccountOwner,
+				ChannelID: feeChannelID,
+			},
+		},
+		validMsg: stakeibctypes.MsgClearBalance{
+			Creator: user.acc.String(),
+			ChainId: HostChainId,
+			Amount:  amount,
+			Channel: feeChannelID,
+		},
+	}
+}
+
+func (s *KeeperTestSuite) TestClearBalance_Successful() {
+	tc := s.SetupClearBalance()
+
+	// Get the sequence number before the ICA is submitted to confirm it incremented
+	feeChannel := tc.initialState.feeChannel
+	feePortId := feeChannel.PortID
+	feeChannelId := feeChannel.ChannelID
+
+	startSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx(), feePortId, feeChannelId)
+	s.Require().True(found, "sequence number not found before reinvestment")
+
+	_, err := s.GetMsgServer().ClearBalance(sdk.WrapSDKContext(s.Ctx()), &tc.validMsg)
+	s.Require().NoError(err, "balance clears")
+
+	// Confirm the sequence number was incremented
+	endSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx(), feePortId, feeChannelId)
+	s.Require().True(found, "sequence number not found after reinvestment")
+	s.Require().Equal(endSequence, startSequence+1, "sequence number after reinvestment")
+}
+
+func (s *KeeperTestSuite) TestClearBalance_HostChainMissing() {
+	tc := s.SetupClearBalance()
+	// remove the host zone
+	s.App.StakeibcKeeper.RemoveHostZone(s.Ctx(), HostChainId)
+	_, err := s.GetMsgServer().ClearBalance(sdk.WrapSDKContext(s.Ctx()), &tc.validMsg)
+	s.Require().EqualError(err, "chainId: GAIA: host zone not registered")
+}
+
+func (s *KeeperTestSuite) TestClearBalance_FeeAccountMissing() {
+	tc := s.SetupClearBalance()
+	// remove the fee account from the host zone
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), HostChainId)
```

Comment:
the nittest: use HZ from `tc.initialState` to limit calling other keeper functions in the test

Reply:
done